### PR TITLE
New action to reload an individual view and all of its children

### LINKED ISSUE #7343: New action to reload an individual view and all of its children
**Is your feature request related to a problem? Please describe.**
There are situations where it is useful to be able to reload the telemetry in an individual view either because it's embedded within a complex layout, or simply to avoid incurring the cost of a browser refresh. An example of this case is when telemetry throttling results in telemetry being dropped. It may be possible to recover the dropped telemetry by re-requesting historical data.

**Describe the solution you'd like**
A new action in a view's menu that triggers an individual view to be reloaded. 

Ideally we would more selectively reload just the telemetry such that we do not lose any ephemeral configuration that the user has defined. This is significantly more work though, and requires new API, modifications to all existing screens, and imposes a new requirement on developers to support it.

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7343 

### Describe your changes:

Adds new action for objects to reload themselves and their children.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        '[aria-label="Data Rate (hz)"]': '0.001'
+      }
+    });
+
+    await page.goto(displayLayout.url);
+
+    // Expand all folders
+    await expandEntireTree(page);
+
+    await page.locator('[title="Edit"]').click();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:64]: We cannot use title's anymore. We should modify this component to have an aria label which matches the title OR make sure it's a button.

This should be `page.getByRole('button' { name: Edit }).click()` or `page.getByLabel('Edit').click()`

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    await page.dragAndDrop(`text='Alpha Table'`, '.l-layout__grid-holder', {
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:66]: "while you're in there" can you update the layout grid holder object to use the appropriate role or aria label?

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      targetPosition: { x: 0, y: 250 }
+    });
+
+    await page.locator('button[title="Save"]').click();
+    await page.getByRole('listitem', { name: 'Save and Finish Editing' }).click();
+  });
+
+  test('can reload display layout and its children', async ({ page }) => {
+    const beforeReloadAlphaTelemetryValue = await page
+      .locator('table.c-telemetry-table__body > tbody')
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:80]: I'm about to implement better locators on telemetry tables. Let's wait for https://github.com/nasa/openmct/pull/7363 to land today

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js (hunk 1 above)
Comment (unlikelyzero) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:64]: https://github.com/nasa/openmct/pull/7363/files#r1446421695

**Comment by scottbell** [comment]
> Let's add a test in openmct-yamcs which inspects the network

@unlikelyzero Can you please make a ticket for that?

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    expect(beforeReloadAlphaTelemetryValue).not.toEqual(afterReloadAlphaTelemetryValue);
+    expect(beforeReloadBetaTelemetryValue).toEqual(afterReloadBetaTelemetryValue);
+
+    // now reload parent
+    await page.getByTitle('More actions').click();
+    await page.getByRole('menuitem', { name: /Reload/ }).click();
+
+    const fullReloadAlphaTelemetryValue = await page
+      .locator('table.c-telemetry-table__body > tbody')
```
[the remark this answers is not in the document]
Comment (scottbell) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:124]: Yeah, tables are pretty nested

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js (hunk 2 above)
Comment (scottbell) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:66]: I think this [needs to be a CSS selector](https://playwright.dev/docs/api/class-page#page-drag-and-drop).

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js (hunk 3 above)
Comment (scottbell) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:80]: @unlikelyzero did this land? What did you want to incorporate here from that PR specifically?

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (scottbell) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js:124]: @unlikelyzero Did you have something specific here to change?

**Comment by scottbell** [comment]
> Code looks good overall. I found an issue where styled components would lose their styles on reload, though. We probably would want to take care of that in this PR?
> Screen.Recording.2024-01-10.at.10.40.02.AM.mov
> Screen.Recording.2024-01-10.at.10.40.14.AM.mov

Nice catch @ozyx! I think I've got that resolved now:

https://github.com/nasa/openmct/assets/9853862/be5bcc65-fba5-4a11-9700-8dac004c60f2

We needed to [re-initialize the style & staleness](https://github.com/nasa/openmct/pull/7362/files#diff-8fcb8316d14608379d1641afa918782dd62db9c6515f5efeac792f21d40edfebR226). I also removed the clear as the `updateView` is already invoking this for us.

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js (hunk 3 above)
Comment (scottbell) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js]: I've updated these to use `aria-labels`

File: e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (scottbell) [reply e2e/tests/functional/plugins/reloadAction/reloadAction.e2e.spec.js]: Switched to using `aria-label`s

**Comment by scottbell** [comment]
> > Let's add a test in openmct-yamcs which inspects the network
> 
> @unlikelyzero Can you please make a ticket for that? We'll need to wait for this to merge before a test over there would work.

Note you currently can't create two branches of the same name for two repositories and have them link to the same ticket in Github Issues. Or at least it doesn't appear to work for me:

https://github.com/nasa/openmct/assets/9853862/aac69d99-506c-435a-85b9-046e3f2ccbb0

File: src/ui/components/ObjectView.vue [hunk 5]
```diff
@@ -222,8 +222,9 @@ export default {
     },
     reload(domainObjectToReload) {
       if (objectUtils.equals(domainObjectToReload, this.domainObject)) {
-        this.clear();
         this.updateView(true);
+        this.initObjectStyles();
```
Comment (ozyx) [inline src/ui/components/ObjectView.vue:226]: nice!

**Review by ozyx — APPROVED** [review]
Great work! LGTM

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**